pandas/core/indexes/base.py

```diff
@@ -1630,8 +1633,11 @@ def to_frame(
         from pandas import DataFrame
 
         if name is lib.no_default:
-            name = self._get_level_names()
-        result = DataFrame({name: self}, copy=False)
+            result_name = self._get_level_names()
+        else:
+            result_name = [name]
+        result = DataFrame(self, copy=False)
+        result.columns = result_name
```

Comment:
Is this type of construction supposed to be equivalent? Not sure if this changed on main but on 2.1.x there are differences if you use a tuple as a column name:

```python
In [6]: tup = ("foo", "bar")

In [7]: pd.DataFrame({tup: [1]})
Out[7]: 
  foo
  bar
0   1
In [8]: df = pd.DataFrame([[1]])

In [9]: df.columns = tup
ValueError: Length mismatch: Expected axis has 1 elements, new values have 2 elements
```

Reply:
Good point. I think this construction is equivalent if we assign `.column` an `Index` instead of an arbitrary iterable. Added a test for this tuple case